### PR TITLE
feat(terminal): change font size and copy/paste using key shortcuts

### LINKED ISSUE #531: Feature Request: Add ability to resize text in xterm.js
I am unable to resize text in the xterm.js instance

I would like to be able to press `ctrl+` to increase text size and `ctrl-` to decrease text size.

### PR DESCRIPTION
**Pull request by zackypick**
Closes #531

### Description

* Added ability to increase/decrease the terminal's font size using keyboard shortcuts CTRL(+)+/- and CTRL(+)0 to reset font size
* Added copy/paste shortcuts using CTRL+SHIFT+(c/v)
* Fixed compilation issues relating to easy-peasy

### Steps to Test

1. Create / start a network
2. Launch a terminal on any LN node
3. Press CTRL(+)+ to increase font size, repeat to further increase until max allowed size
4. Press CTRL(+)- to decrease font size, repeat to further increase until min allowed size
5. Press CTRL(+)0 reset terminal's font size
6. Select some text and press CTRL(+)SHIFT(+)c to copy text to clipboard
7. Press CTRL(+)SHIFT(+)v to paste text from clipboard

### Screenshots

image

### CODE REVIEW

**Comment by jamaljsr** [comment]
Thanks for the PR. I'll try to review this later today or tomorrow at the latest.

**Comment by jamaljsr** [comment]
Can you rebase this on master. I noticed that there are a lot of dependency version downgrades in the package.json file.

**Comment by zackypick** [comment]
> Can you rebase this on master. I noticed that there are a lot of dependency version downgrades in the package.json file.

Hi Jamal, you're right. the thing is the project does not compile with the master branch's dependencies (I assume the bot upgraded lib versions which breaks things?)

File: src/components/terminal/DockerTerminal.tsx
Comment (jamaljsr) [inline src/components/terminal/DockerTerminal.tsx:212]: The `.toString()` isn't necessary since you are defaulting to an empty string.

File: src/utils/tests/renderWithProviders.tsx
Comment (jamaljsr) [inline src/utils/tests/renderWithProviders.tsx:89]: These overrides shouldn't be necessary.

File: src/components/terminal/DockerTerminal.tsx
Comment (jamaljsr) [inline src/components/terminal/DockerTerminal.tsx:165]: You can use `if (terminal?.current?.options.fontSize)` instead to simplify the condition.

File: src/components/terminal/DockerTerminal.tsx
Comment (jamaljsr) [inline src/components/terminal/DockerTerminal.tsx:184]: After changing the font size, you should call `fitAddon.fit();` like on line 256 to prevent this background color mismatch. You'll of course need to move the declaration of `fitAddon` to the component level scope to use it here.

image

**Review by jamaljsr — CHANGES_REQUESTED** [review]
Thanks so much for the contribution. I left my feedback below.

File: src/components/terminal/DockerTerminal.tsx
Comment (zackypick) [reply src/components/terminal/DockerTerminal.tsx:213]: "Nice touch adding the info message on copy" - Thank you :-) 

The trailing then() was to just keep the compiler happy with the promise, but not really necessary. Removed it.

File: src/components/terminal/DockerTerminal.tsx
Comment (zackypick) [reply src/components/terminal/DockerTerminal.tsx:200]: Great point! thank you, done.

File: src/components/terminal/DockerTerminal.tsx
Comment (zackypick) [reply src/components/terminal/DockerTerminal.tsx:212]: Good point. Done.

File: src/components/terminal/DockerTerminal.tsx
Comment (zackypick) [reply src/components/terminal/DockerTerminal.tsx:165]: Of course. Somehow it was showing an error for me earlier when using optional chaining. Fixed now.

**Comment by zackypick** [comment]
Done, squashed into a single commit. Hope it can now be merged

**Comment by jamaljsr** [comment]
The diff looks great. I'm going to do a final sanity check on Mac/Win/Linux before merging. Hopefully will get that done tonight or tomorrow.

**Comment by jamaljsr** [comment]
Hey, so sorry but I just realized the Polar asciiart is now out of alignment. It looks like you removed the whitespace on [line 45](https://github.com/jamaljsr/polar/pull/612/files#diff-08022456b4d5e33a2914504f5bd99eed0a5b699a7fe96dec1d67e9dbd919e83aR45). Can you fix this please?

image

**Comment by jamaljsr** [comment]
Hey, so sorry but I just realized the Polar asciiart is now out of alignment. It looks like you removed the whitespace on [line 45](https://github.com/jamaljsr/polar/pull/612/files#diff-08022456b4d5e33a2914504f5bd99eed0a5b699a7fe96dec1d67e9dbd919e83aR45). Can you fix this please?

image

**Comment by jamaljsr** [comment]
BTW: I tested on all OS's so this is good to go once the whitespace is fixed.

**Comment by zackypick** [comment]
No problem, fixed. For some reason, VSCode automatically removed this trailing newline. Anyway hope everything is okay now.

**Review by jamaljsr — APPROVED** [review]
This looks good to go. 👍 Thanks for your contribution.

**Comment by zackypick** [comment]
💯

**Pull request merged**
